src/api/firebase.js

```diff
@@ -170,25 +171,23 @@ export async function shareList(listPath, currentUserId, recipientEmail) {
  */
 export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listPath, 'items');
-	// TODO: Replace this call to console.log with the appropriate
-	// Firebase function, so this information is sent to your database!
 	return addDoc(listCollectionRef, {
 		dateCreated: new Date(),
-		// NOTE: This is null because the item has just been created.
-		// We'll use updateItem to put a Date here when the item is purchased!
 		dateLastPurchased: null,
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath, itemId) {
+	const itemDocRef = doc(db, listPath, 'items', itemId);
+	const datePurchased = new Date();
+	await updateDoc(itemDocRef, {
```

Comment:
We should wrap `updateDoc` in a try/catch for handling errors that may occur during the update operation, similar to how we've done [here](https://github.com/the-collab-lab/tcl-75-smart-shopping-list/blob/1eb1aea25080c75518be7770f817675dee34d132/src/api/firebase.js#L156)! 🚀